runtime/vm_hooks.go

```diff
@@ -0,0 +1,76 @@
+package runtime
+
+import (
+	"github.com/d5/tengo/compiler"
+	"github.com/d5/tengo/compiler/source"
+	"github.com/d5/tengo/objects"
+)
+
+type hooks struct {
```

Comment:
can VM itself implement the interface?

```golang
func (v *VM) Call(val objects.Object, args ...objects.Object) (objects.Object, error) {
   // ...
}
```

Reply:
I chose this approach because I thought it would be better if end users couldn't type-assert `RuntimeHooks` to `*runtime.VM`, but willing to change if you feel like that's an unnecessary protection